### PR TITLE
Refactored Tools::getCountry function

### PR DESCRIPTION
**Pull request by michaelKaefer**
Method was not commented or explained, did my best to refactor. Changed method definition -> deprecated

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Method was not commented or explained, did my best to refactor. Changed method definition -> deprecated
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | yes
| Fixed ticket? | -
| How to test?  | no existing test found

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9060)

### CODE REVIEW

**Review by mickaelandrieu — COMMENTED** [review]
looks good but I don't feel confident, let's wait for one more review :)

**Comment by Quetzacoalt91** [comment]
When we refacto methods and classes, the previous behavior should be covered by unit tests. Then these tests will confirm nothing has changed between the old and the new version.

I'd be confident in approving this PR if the tests were provided at the same time.

File: classes/Tools.php
Comment (Quetzacoalt91) [inline classes/Tools.php]: I dislike the affectation in the test. What do you think about

```php
        if (null !== $address && $address->id_country) {
            return $address->id_country;
        }
```

?

File: classes/Tools.php
Comment (Quetzacoalt91) [inline classes/Tools.php]: It seems in the previous version, that case can be overridden if `Configuration::get('PS_DETECT_COUNTRY')` is true. In my opinion the behavior is changing here (and this explain why I was talking about tests)

File: classes/Tools.php
Comment (michaelKaefer) [reply classes/Tools.php]: @Quetzacoalt91 Thank you! I made another commit.

File: classes/Tools.php
Comment (mickaelandrieu) [reply classes/Tools.php]: @michaelKaefer if you don't know how to write tests with PHPUnit and you want to try, we can help you to help us 🤣

**Comment by michaelKaefer** [comment]
@Quetzacoalt91 @mickaelandrieu I commited a test. Yes, I definitly need your help. I can't even figure out what this method should do.. If it really works correctly in its old state, of course I can try to refactor it to produce the same results. Can you approve that the old method works correctly after having a look on the test results?

**Comment by mickaelandrieu** [comment]
@michaelKaefer I didn't know this class was tested o_O 

I think we're good here then, thank you!

**Review by mickaelandrieu — DISMISSED** [review]

**Comment by mickaelandrieu** [comment]
wdyt @Quetzacoalt91 ? at least;, it's tested now

**Comment by PierreRambaud** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[tests/Unit/Classes/ToolsCoreTest.php, line 434 at r3](https://reviewable.io/reviews/prestashop/prestashop/9060#-LCgmucB5s_KrJ15Yo_V:-LCgmucC11McEGjysVUG:b-jlh4ww) ([raw file](https://github.com/prestashop/prestashop/blob/9a7c032954c98fd1839d462d2b9825493e821f63/tests/Unit/Classes/ToolsCoreTest.php#L434)):*
> ```PHP
> 
>         // Set dependencies
>         $address = new \Address();
> ```

Can you use namespace please for `Address`, `Configuration` and `Tools`?

---


*Comments from [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9060)*

**Comment by michaelKaefer** [comment]
@mickaelandrieu @Quetzacoalt91 The test only works if in your DB there is a country with id=21 and iso-code=EN. Otherwise the test is senseless. Is this guaranteed by the testing setup we use? So the refactor returns now exactly the same as before, I fixed the refactor. But the test is a problem if there is no such country as described. @PierreRambaud I added the requested use statements.

**Comment by PierreRambaud** [comment]
@mickaelandrieu Should be ok, wdyt?

**Comment by michaelKaefer** [comment]
@PierreRambaud @Progi1984 I'll close this. I think you got enough work to do.

**Comment by matks** [comment]
> @PierreRambaud @Progi1984 I'll close this. I think you got enough work to do.

Thanks for the time you gave to this project 🤝

**Pull request closed without merging**
